.github/workflows/test_android.yml

```diff
@@ -0,0 +1,59 @@
+name: 🧪 Android tests
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  test-android:
+    name: Test Android
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          submodules: false # don't initialize submodules automatically
+
+      - name: Prepare testing
+        uses: ./.github/actions/prepare-testing
+        timeout-minutes: 15
+        with:
+          arch: x86_64
+          android: true
+
+      # Needed for Android emulator
+      - name: Enable KVM
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
+
+      # Make sure that the required directories and .cfg do exists. Workaround to keep ADV happy on `ubuntu-latest`.
+      - name: Setup Android directories
+        run: |
+          mkdir -p $HOME/.android
+          mkdir -p $HOME/.android/avd
+          touch $HOME/.android/repositories.cfg
+
+      - name: Run Android Tests
+        uses: reactivecircus/android-emulator-runner@b530d96654c385303d652368551fb075bc2f0b6b # pin@v2.35.0
+        timeout-minutes: 30
+        with:
+          api-level: 30
```

Comment:
Afaik Google Play requires existing apps to target at least API level 34 to remain available to new users. Should we bump it here as well?